packages/userstorage/src/browser/user-storage-service-filesystem.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { Disposable, DisposableCollection, ILogger, Emitter, Event } from '@theia/core/lib/common';
+import { UserStorageChangeEvent, UserStorageService } from './user-storage-protocol';
+import { injectable, inject } from 'inversify';
+import { FileSystem } from '@theia/filesystem/lib/common';
+import { FileSystemWatcherServer, DidFilesChangedParams } from '@theia/filesystem/lib/common/filesystem-watcher-protocol';
+import * as jsoncparser from 'jsonc-parser';
+import URI from '@theia/core/lib/common/uri';
+import { UserStorageUri } from './user-storage-uri';
+
+@injectable()
+export class UserStorageServiceFilesystem implements UserStorageService {
```

Comment:
This class should be named `UserStorageService`, you don't need the interface

Reply:
We will have a second implementation, that doesn't use the file system, but a central workspace management server (e.g. Che) that contains user data in a database.

Reply:
ok, sorry, overlooked the comment from @epatpol above, but `FileSystemUserStorageService` sounds better to me than `UserStorageServiceFilesystem`.

Reply:
I would prefer `<What><How>Iml`, i.e. `UserStorageServiceFileSystemImpl`.

Reply:
As `UserStorageServiceFilesystem` was already an `Impl` of `UserStorageService` I ometted it, but now I'm more confused.

Reply:
Sorry, It is just a pattern I usually use when having different implementations of an interface I suffix the class with `Impl` instead of prefixing the interface with `I`. And before the `Impl` I put the kind of implementation, e.g. in this case
 - `UserStoraageServiceFileSystemImpl` (based on user home)
 - `UserStorageServiceWorkspaceServerImpl` (based on central workspace Server)

I think we kind of agreed on that here https://github.com/theia-ide/theia/wiki/Coding-Guidelines#interfaces

Reply:
Thanks @svenefftinge, I've changed the name for `UserStorageServiceFileSystemImpl`.